components/event-bus/test/integration/publish_push_test.go

```diff
@@ -127,209 +87,86 @@ func Test_Subscriber_Status(t *testing.T) {
-		log.Printf("%v", respObj)
-
-		var ok bool
-		for i := 0; i < 10; i++ {
-			time.Sleep(1 * time.Second)
-			res, err := http.Get(subscriberServerV1.URL + util.SubServer1ResultsPath)
-			assert.Nil(t, err)
-			body, err := ioutil.ReadAll(res.Body)
-			var resp string
-			json.Unmarshal(body, &resp)
-			res.Body.Close()
-			if len(resp) == 0 {
-				continue
-			}
-			assert.Equal(t, eventDataV1, resp)
-			ok = true
-			break
-		}
-		assert.True(t, ok)
-	}
+	verifyPublishPushFlow("Test_Publish_Push_Request_1", "sub-1", "namespace-1", t)
 
-	{
-		payloadV2 := makePayload(sourceIDV2, eventType, eventTypeVersion, eventDataV2)
-		res, err := http.Post(publishServer.URL+"/v1/events", "application/json", strings.NewReader(payloadV2))
-		checkIfError(err, t)
-		verifyStatusCode(res, 200, t)
-		log.Print(res)
-		respObj := &api.PublishResponse{}
-		body, err := ioutil.ReadAll(res.Body)
-		defer res.Body.Close()
-		err = json.Unmarshal(body, &respObj)
-		assert.NotNil(t, respObj.EventID)
-		assert.NotEmpty(t, respObj.EventID)
-		log.Printf("%v", respObj)
-
-		var ok bool
-		for i := 0; i < 10; i++ {
-			time.Sleep(1 * time.Second)
-			res, err := http.Get(subscriberServerV2.URL + util.SubServer2ResultsPath)
-			assert.Nil(t, err)
-			body, err := ioutil.ReadAll(res.Body)
-			var resp string
-			json.Unmarshal(body, &resp)
-			res.Body.Close()
-			if len(resp) == 0 {
-				continue
-			}
-			assert.Equal(t, eventDataV2, resp)
-			ok = true
-			break
-		}
-		assert.True(t, ok)
-	}
+	verifyPublishPushFlow("Test_Publish_Push_Request_2", "sub-2", "namespace-1", t)
 }
 
-func newFakeInformer() cache.SharedIndexInformer {
-	sub := util.NewSubscription(
-		"test-sub",
-		metav1.NamespaceDefault,
-		subscriberServerV1.URL+util.SubServer1EventsPath,
-		eventType,
-		eventTypeVersion,
-		sourceIDV1)
-	clientSet := fake.NewSimpleClientset(sub)
-	informer := v1alpha1.NewSubscriptionInformer(clientSet, metav1.NamespaceAll, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
-	informer.GetIndexer().Add(sub)
-	return informer
-}
+func Test_sameSubjectSubscribersInDifferentNamespacesShouldReceiveEventsOfThatSubject(t *testing.T) {
+	// create two subscriptions with the same name in two different namespaces to the same subject
+	name := "test-subscription"
+	ns1 := "namespace-1"
+	ns2 := "namespace-2"
+	eventData := "Test_sameSubjectSubscribersInDifferentNamespacesShouldReceiveEventsOfThatSubject"
 
-func verifyStatusCode(res *http.Response, expectedStatusCode int, t *testing.T) {
-	if res.StatusCode != expectedStatusCode {
-		t.Errorf("Status code is wrong, have: %d, want: %d", res.StatusCode, expectedStatusCode)
-	}
-}
+	doCreateSub(name, ns1, subscriberServerV1.URL+util.SubServer1EventsPath, eventType, eventTypeVersion, sourceIDV1, t)
 
-func checkIfError(err error, t *testing.T) {
-	if err != nil {
-		t.Fatal(err)
-	}
+	doCreateSub(name, ns2, subscriberServerV2.URL+util.SubServer2EventsPath, eventType, eventTypeVersion, sourceIDV1, t)
```

Comment:
"do" as a prefix is not really used in Go, I'll prefer the simple function name "createSub"